R/fmelt.R

```diff
@@ -31,8 +33,23 @@ melt.data.table <- function(data, id.vars, measure.vars, variable.name = "variab
     measure.vars = patterns(pats, cols=cols)
   }
   if (is.list(measure.vars) && length(measure.vars) > 1L) {
-    if (length(value.name) == 1L)
-      value.name = paste(value.name, seq_along(measure.vars), sep="")
+    meas.nm = names(measure.vars)
+    if (is.null(meas.nm)) {
+      # user-provided or default stub
+      if (length(value.name) == 1L) {
```

Comment:
@MichaelChirico no idea if this can make some problem but what if `meas.nm` is NULL and `length(value.name) > 1L`? this case is not covered here.